uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java

```diff
@@ -690,127 +697,127 @@ public synchronized void handleLiveInstanceChange(boolean onlyCheckOffline, bool
       List<String> liveInstances = HelixUtils.liveInstances(_helixManager);
       List<String> instanceToReplace = new ArrayList<>();
       boolean routeControllerDown = false;
-      for (String instanceName : instanceToTopicPartitionsMap.keySet()) {
-        if (!liveInstances.contains(instanceName)) {
-          routeControllerDown = true;
-          instanceToReplace.add(instanceName);
+      // Check if any worker in route is down
+      boolean routeWorkerDown = false;
+      if (_enableRebalance || forceBalance) {
```

Comment:
nit: it can reduce nested level (and would be easier to read), if you can write sth like
```
if (!(_enableRebalance || forceBalance)) return
```

Reply:
there is autoscaling follow by autobalancing, so we could't return if auto balance is disabled .
I'm going to extract controller auto balancing and worker auto balancing logic separate function, this should help the readability